client/packages/programs/src/JsonForms/common/components/Array.tsx

```diff
@@ -254,8 +255,47 @@ const ArrayComponent = (props: ArrayControlCustomProps) => {
     label,
     rootSchema,
     renderers,
+    config,
   } = props;
 
+  const isElementEditable = (child: any, index: number) => {
+    if (!enabled) return false;
+    if (!uischema.options?.['editRestrictions']) return true;
+
+    const restrictions = uischema.options['editRestrictions'];
+
+    /*
+    The "editRestrictions" option is an object containing any number of the
+    following properties:
+    - "latest": (true/false) -- Only the latest array element can be
+      edited/removed
+    - "authorId": (true/false) --  The element's data.authorId field must match
```

Comment:
the naming of `authorId` being a bool field is odd to me.
something like `isCurrentUser` makes more sense - I was expecting 'authorId' to be a string prop, not a bool

I don't see the need to be so restrictive - and I would allow a user to edit all of their own notes ( i.e. remove `maxAge` and `latest` )

Reply:
> the naming of `authorId` being a bool field is odd to me.
> something like `isCurrentUser` makes more sense - I was expecting 'authorId' to be a string prop, not a bool

Done :)

> I don't see the need to be so restrictive - and I would allow a user to edit all of their own notes ( i.e. remove `maxAge` and `latest` )

Have left them in for now for demonstration purposes, but I'll pull them out of the uischema before merging (i.e. the options will still exist, but we won't use them (for now). I suspect there will be ongoing discussion about what the "rules" for these are.